### ISSUE TITLE
postgresql96 on 10.14

### ISSUE THREAD

**Issue by TheSin-** [fink/fink-distributions#291]
To get postgresql96 to build on 10.14, there is a perl module, so we need to build the perl module in a separate package using the already patched fink, or add

```
export PERLINC=`xcrun --sdk macosx --show-sdk-path`/System/Library/Perl/`perl -e 'print substr($^V, 1)' | cut -d. -f1 -f2`/darwin-thread-multi-2level/CORE

CPPFLAGS="${CPPFLAGS} -I${PERLINC}" ./configure \
```

to the postgresql96 CompileScript section.

Also on my i9 MBP I hit a race condition in the build process, so MAXJOBS might need to be reduced

**Comment by nieder** [comment]
postgresql96 belongs to @wodev

**Comment by wodev** [comment]
There is also a Bug Report in the pgsql-bugs Mailing List

[BUG #15358: PostgreSQL fails to build on 10.14 when Perl is enabled.](https://www.postgresql.org/message-id/153558865647.1483.573481613491501077%40wrigleys.postgresql.org)

I'll update the Postgresql packages in the next few days.

**Comment by remkos** [comment]
@nieder: @TheSin- did not reply, but I can tell you that I still see the same as he does.
I do not think that #314 and this one have the same scope.
The need here is to find "perl.h", given this error message during the `configure` step.
<pre>
checking for perl.h... no
configure: error: header file <perl.h> is required for Perl
</pre>

**Comment by remkos** [comment]
Thanks for the response. But it does not work for me. I suspect this is because I do not have Xcode 10.1 but the Xcode Command Line Tools 10.1 install. I guess that the path to `perl.h` is different between the two. For me it is: `/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/System/Library/Perl/5.18/darwin-thread-multi-2level/CORE/perl.h`
This would be properly captured by the `xcrun` command specified by @TheSin- above:
<pre>
$ xcrun --sdk macosx  --show-sdk-path
/Library/Developer/CommandLineTools/SDKs/MacOSX10.14.sdk
</pre>
I looked at the differences upstream between postgresql 9.6.10 and 9.6.11 and there is really nothing there that supports this path. So I suspect we really need addition to `FCFLAGS` suggested by @TheSin-

**Comment by remkos** [comment]
PS: @wodev: Can you check where your `perl.h` is found?

**Comment by wodev** [comment]
The only Location for perl.h is:

The path is resolved with the clang sysroot flag:
`-isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk`

And perl.h is in:
`/Applications/Xcode.app//Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/System/Library/Perl/5.18/darwin-thread-multi-2level/CORE/perl.h`

And the fix in 9.6.11 is the postgresql commit 0a4456a .

**Comment by remkos** [comment]
That was very helpful, @wodev.
Now I understand why it does not run for me, nor for @TheSin-. We both installed Xcode Command Line Tools only (so not the full Xcode). Obviously the path you have is not on our MacOS 10.14.

I found the (new) line in postgresql commit [0a4456a](https://github.com/postgres/postgres/commit/0a4456a70dfa004b2904c6a315dd32f044336bcc) that is supposed to get the right path:
<pre>
PG_SYSROOT=`xcodebuild -version -sdk macosx Path 2>/dev/null`
</pre>
But this **does not** work with the Command Line Tools. That command results in:
<pre>
$ xcodebuild -version -sdk macosx Path
xcode-select: error: tool 'xcodebuild' requires Xcode, but active developer directory '/Library/Developer/CommandLineTools' is a command line tools instance
</pre>
That's why @TheSin- suggested the following command instead:
<pre>
$ xcrun --sdk macosx --show-sdk-path
/Library/Developer/CommandLineTools/SDKs/MacOSX10.14.sdk
</pre>
with the result printed above.

@wodev, can you verify that you get the right path using Xcode with the `xcron` command, i.e. `/Applications/Xcode.app//Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk`?

Then we should either include the hack suggested by @TheSin- to the `.info` file(s) and/or suggest this patch to the postgresql developers, as their update for 10.14 would **only** work with Xcode, and not with the Command Line Tools. You can help them confirm that the `xcrun` command provides the right path in either case.

Thanks!

**Comment by wodev** [comment]
@remkos  thanks for your analysis.

I also get the correct SDK Path with `xcrun`  and the full Xcode installation.

```
xcrun --sdk macosx --show-sdk-path
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk
```

PR #332 contains a patch to use `xcrun` instead of `xcodebuild`.  So it should the isysroot path correctly and detect perl.h on 10.14 for Xcode command line tool only installations.

**Comment by remkos** [comment]
Closed by the implementation of PR #332.
